common/beerocks/hostapd/configuration.cpp

```diff
@@ -0,0 +1,138 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2016-2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include "configuration.h"
+#include <algorithm>
+#include <easylogging++.h>
+#include <fstream>
+
+namespace prplmesh {
+namespace hostapd {
+namespace config {
+
+Configuration::Configuration(const std::string& file_name) : m_configuration_file(file_name) {}
+
+Configuration::operator bool() const { return m_ok; }
+
+bool Configuration::load()
+{
+    // please take a look at the end of this file for
+    // the expected format of hostapd configuration file
+    // loading the file relays on the expected format
+    // otherwise the load fails
```

Comment:
as mentioned before - this is exactly why you should move the how-to-use to a separate .txt or .md file - and here just mention `see how-to-file-name.md for expected configuration format`
```suggestion
    // please take a look at the end of this file for
    // the expected format of hostapd configuration file
    // loading the file relies on the expected format
    // otherwise the load fails
```